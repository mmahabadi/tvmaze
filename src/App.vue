<template>
  <RouterView />
</template>

<script setup lang="ts">
import { provide } from 'vue'
import { RouterView } from 'vue-router'
import { type AppSettings } from '@/types'

provide('appSettings', {
  itemsInEachGenreSection: 20,
  itemsInEachPage: 2,
  itemsToShowInHomePageSlider: 15,
  yearLimitToConsiderAsRecentToShowInHomePageSlider: 2020,
  toasterTimeout: 5000,
} as AppSettings)
</script>
