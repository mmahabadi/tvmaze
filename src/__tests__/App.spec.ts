import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import App from '../App.vue'

describe('App', () => {
  it('renders correctly', () => {
    const wrapper = mount(App, {
      global: {
        stubs: {
          RouterView: true
        }
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
