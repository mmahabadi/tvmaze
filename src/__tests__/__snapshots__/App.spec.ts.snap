// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > renders correctly 1`] = `"<router-view-stub name="default"></router-view-stub>"`;
